src/app/[locale]/viewer/_components/SubmodelsOverviewCard.tsx

```diff
@@ -4,35 +4,45 @@ import { useIsMobile } from 'lib/hooks/UseBreakpoints';
 import { messages } from 'lib/i18n/localization';
 import { FormattedMessage } from 'react-intl';
 import { SubmodelDetail } from './submodel/SubmodelDetail';
-import { Key, Reference, Submodel } from '@aas-core-works/aas-core3.0-typescript/types';
+import { Reference, Submodel } from '@aas-core-works/aas-core3.0-typescript/types';
 import { useAsyncEffect } from 'lib/hooks/UseAsyncEffect';
 import { SubmodelSorting } from 'app/[locale]/viewer/_components/submodel/sorting/SubmodelSorting';
 import { TabSelectorItem, VerticalTabSelector } from 'components/basics/VerticalTabSelector';
 import { MobileModal } from 'components/basics/MobileModal';
 import { useApis } from 'components/azureAuthentication/ApiProvider';
 import { useRegistryAasState } from 'components/contexts/CurrentAasContext';
 import { getSubmodelFromSubmodelDescriptor } from 'lib/searchUtilActions/search';
+import { useEnv } from 'app/env/provider';
+
 
 export type SubmodelsOverviewCardProps = { readonly smReferences?: Reference[]; readonly isLoading?: boolean };
 
 export function SubmodelsOverviewCard(props: SubmodelsOverviewCardProps) {
     const [selectedItem, setSelectedItem] = useState<TabSelectorItem>();
     const [selectedSubmodel, setSelectedSubmodel] = useState<Submodel>();
     const { submodelClient } = useApis();
-    const [ registryAasData] = useRegistryAasState();
+    const [registryAasData] = useRegistryAasState();
+    const { submodelRegistryServiceClient } = useApis();
 
     SubmodelSorting(selectedSubmodel);
 
     const [items, setItems] = useState<TabSelectorItem[]>([]);
     const [open, setOpen] = useState<boolean>(false);
     const isMobile = useIsMobile();
     const firstSubmodelIdShort = 'Nameplate';
+    const env = useEnv();
 
     useAsyncEffect(async () => {
         if (!props.smReferences) return;
 
         const submodels: { id: string; label: string; metadata?: Submodel; endpoint?: string }[] = [];
 
+        async function fetchSubmodelFromRepo(reference: Reference) {
+            const id = reference.keys[0].value;
+            const metadata = await submodelClient.getSubmodelMetaDataById(id);
```

Comment:
Unsure if my PR #30 will change this as well, please let us check this when you merge